### ISSUE TITLE
SmartGit is not starting, crashes with error "Could not create transient scope for PID"

### ISSUE THREAD

**Issue by MetalArend** [NixOS/nixpkgs#185830]
### Describe the bug
Trying to launch SmartGit ends with the following error in `journalctl -f`:

> nixos .gnome-shell-wr[18136]: Could not create transient scope for PID 22233: GDBus.Error:org.freedesktop.DBus.Error.UnixProcessIdUnknown: Process with ID 22233 does not exist.

### Steps To Reproduce
1. Install smartgithg with home-manager
2. Try to start smartgithg

### Expected behavior
SmartGit coming to life in a new window

### Notify maintainers
@jraygauthier @MinerSebas 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.06 MiB download, 0.30 MiB unpacked):
  /nix/store/9m5k22zdy4s1yh003ma0igqmmrvncww1-bash-interactive-5.1-p16-dev
copying path '/nix/store/9m5k22zdy4s1yh003ma0igqmmrvncww1-bash-interactive-5.1-p16-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.59, NixOS, 22.05 (Quokka), 22.05.2322.92fe622fdfe`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.1`
 - channels(mammi): `"home-manager-22.05.tar.gz"`
 - channels(root): `"nixos-22.05"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by MetalArend** [comment]
So, after a bit more investigating, it seems that smartgit does not install a jre package if it is missing. After looking into the bin directory, it seems there is a .smartgit-wrapped script, that failed with the error that jre 11 is missing. After installing that through home-manager, at least the .smartgit-wrapped script is able to start through the commandline. However, the link available in the Applications overview is not opening anything. Could it have something to do with a nixos upgrade from 21.05 to 22.05?

**Comment by hab25** [comment]
I suspect the problem is with `smartgithg`'s usage of  `wrapGAppsHook`, i.e. between https://github.com/NixOS/nixpkgs/blob/b70a718f7f04358f6ca5454d75f3e9d5153ae6f3/pkgs/applications/version-management/smartgithg/default.nix#L22 and https://github.com/NixOS/nixpkgs/blob/b70a718f7f04358f6ca5454d75f3e9d5153ae6f3/pkgs/applications/version-management/smartgithg/default.nix#L38

I'm not familiar with how `wrapGappsHook` works and don't have enough `nix` knowledge to understand it and produce a fix quickly, but with the following workaround/hack everything works for me, including the `.desktop` file. 
```nix
      # workaround for https://github.com/NixOS/nixpkgs/issues/185830 todo remove this when the issue is fixed
      (
        let
          sg = pkgs.smartgithg;
        in
          pkgs.runCommand "${sg.name}-hab25-workaround" {} ''
            mkdir $out
            ln --symbolic ${sg}/* $out
            rm $out/bin
            mkdir $out/bin
            newExe="$out/bin/smartgit"
            echo "\
              #!${pkgs.lib.getExe pkgs.bash}
              export PATH=${pkgs.jre}/bin:\$PATH
              sh ${sg}/bin/.smartgit-wrapped \
              " > $newExe
            chmod +x $newExe
          ''
      )
```
If you wish to use this, make sure to make the above package, but not `pkgs.smartgithg`,  available in your environment's `$PATH`, so that you don't face ambiguity problems.

**Comment by MetalArend** [comment]
I've given up on NixOS, it's way too much trouble if you're using tools like this. But I'll keep this issue open, in the hopes others that want to use SmartGit might one day be able to do so.

**Comment by hab25** [comment]
@MetalArend I fully agree that NixOS is currently not worth it as a workstation. For example, the applications that most matter to me are google chrome, intellij idea, the android development tools, logseq and copyq. I think this is quite a minimalist selection, but even so, every single respective nixpkg has currently open usability issues, apart from copyq: 
https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+%22google+chrome%22+OR+%22intellij%22+OR+%22android-studio%22+OR+%22jetbrains%22+OR+%22copyq%22+OR+%22logseq%22+sort%3Areactions-%2B1-desc+label%3A%220.kind%3A+bug%22+is%3Aopen

If you haven't already, you might want to try [`home-manager` ](https://github.com/nix-community/home-manager) on another GNU/Linux distro. It makes it easy for you to use `nix` incrementally while still being able to fallback to that distro's package manager (or flatpak, appimage, building from source, snap, etc) when you encounter critically broken packages like this.

**Comment by MinerSebas** [comment]
This issue was fixed by #197927 (which is part of the 22.11 release) and can now be closed.
